File: PHP/src/Jakopec/MjesanjeTriUnosa.php
<?php


namespace Jakopec;


use Exception;

class MjesanjeTriUnosa extends MjesanjeDvaUnosa
{

    private $bodovaTreciUnos;
    private $zvanjeTreciUnos;

    /**
     * MjesanjeDvaUnosa constructor.
     * @param int $id
     * @param bool $stiglja
     * @param bool $belot
     * @param DateTime|null $datumUnosa
     * @param int $bodovaPrviUnos
     * @param int $bodovaDrugiUnos
     * @param int $zvanjePrviUnos
     * @param int $zvanjeDrugiUnos
     * @param int $bodovaTreciUnos
     * @param int $zvanjeTreciUnos
     * @throws Exception
     */
    public function __construct(int $id=0, bool $stiglja = false, bool $belot = false, DateTime $datumUnosa = null, int $bodovaPrviUnos = 0, int $bodovaDrugiUnos = 0, int $zvanjePrviUnos = 0, int $zvanjeDrugiUnos = 0, int $bodovaTreciUnos = 0, $zvanjeTreciUnos = 0)
    {
        parent::__construct($id, $stiglja, $belot, $datumUnosa, $bodovaPrviUnos, $bodovaDrugiUnos, $zvanjePrviUnos, $zvanjeDrugiUnos);
        $this->bodovaTreciUnos = $bodovaTreciUnos;
        $this->zvanjeTreciUnos = $zvanjeTreciUnos;
    }

    /**
     * @inheritDoc
     */
    public function getRezultat(): Rezultat
    {
        $rezultat = parent::getRezultat();
        $rezultat->setTreci($this->getBodovaTreciUnos() + $this->getZvanjeTreciUnos());
        return $rezultat;
    }

    /**
     * @return int
     */
    public function getBodovaTreciUnos(): int
    {
        return $this->bodovaTreciUnos;
    }

    /**
     * @param int $bodovaTreciUnos
     */
    public function setBodovaTreciUnos(int $bodovaTreciUnos)
    {
        $this->bodovaTreciUnos = $bodovaTreciUnos;
    }

    /**
     * @return int
     */
    public function getZvanjeTreciUnos(): int
    {
        return $this->zvanjeTreciUnos;
    }

    /**
     * @param int $zvanjeTreciUnos
     */
    public function setZvanjeTreciUnos(int $zvanjeTreciUnos)
    {
        $this->zvanjeTreciUnos = $zvanjeTreciUnos;
    }
}

File: Swift4/Swift4/model/MjesanjeTriUnosa.swift
//
//  MjesanjeDvaUnosa.swift
//  Swift4
//
//  Created by Tomislav Jakopec on 22/02/2020.
//  Copyright © 2020 Tomislav Jakopec. All rights reserved.
//

import Foundation
class MjesanjeTriUnosa: MjesanjeDvaUnosa{
    
    private var bodovaTreciUnos: Int
    private var zvanjeTreciUnos: Int
    
    
   
    
    init(sifra: Int = 0, stiglja: Bool = false, belot: Bool = false, datumUnosa: Date = Date.init(), bodovaPrviUnos: Int = 0, zvanjePrviUnos: Int = 0, bodovaDrugiUnos: Int = 0, zvanjeDrugiUnos: Int = 0, bodovaTreciUnos: Int = 0, zvanjeTreciUnos: Int = 0) {
        self.bodovaTreciUnos=bodovaTreciUnos
        self.zvanjeTreciUnos=zvanjeTreciUnos
        super.init(sifra: sifra, stiglja: stiglja, belot: belot, datumUnosa: datumUnosa,bodovaPrviUnos: bodovaPrviUnos, zvanjePrviUnos: zvanjePrviUnos, bodovaDrugiUnos: bodovaDrugiUnos, zvanjeDrugiUnos: zvanjeDrugiUnos)
    }
    
    public func getBodovaTreciUnos()->Int{
        return self.bodovaTreciUnos
    }
    
    public func setBodovaTreciUnos(bodovaTreciUnos: Int){
        self.bodovaTreciUnos = bodovaTreciUnos
    }
    
    public func getZvanjeTreciUnos()->Int{
        return self.zvanjeTreciUnos
    }
    
    public func setZvanjeTreciUnos(zvanjeTreciUnos: Int){
        self.zvanjeTreciUnos = zvanjeTreciUnos
    }
    
    
    override func getRezultat()->Rezultat{
        let rezultat = super.getRezultat()
        rezultat.setTreci(treci: getBodovaTreciUnos() + getZvanjeTreciUnos())
        return rezultat
    }
    
    
    
}
